lldb/source/Plugins/ObjectFile/Minidump/MinidumpFileBuilder.cpp

```diff
@@ -657,8 +657,11 @@ MinidumpFileBuilder::AddMemoryList(const lldb::ProcessSP &process_sp,
     auto data_up = std::make_unique<DataBufferHeap>(size, 0);
     const size_t bytes_read =
         process_sp->ReadMemory(addr, data_up->GetBytes(), size, error);
-    if (bytes_read == 0)
+    if (error.Fail() || bytes_read == 0) {
```

Comment:
Instead of silently swallow the failure, we should emit the error message to terminal/console so that users are aware of region skipping.

Reply:
Would that be too chatty? Maybe a warning msg if there's any and use log channel for more details?

Reply:
So a few things here:
1 - We should fix `Process::CalculateCoreFileSaveRanges(...)` to not include memory ranges that have no read access. The fix will need to go into this function in Process.cpp:
```
static void AddRegion(const MemoryRegionInfo &region, bool try_dirty_pages, Process::CoreFileMemoryRanges &ranges) {
  // Don't add empty ranges or ranges with no permissions.
  if (region.GetRange().GetByteSize() == 0 || region.GetLLDBPermissions() == 0)
    return;
  if (try_dirty_pages && AddDirtyPages(region, ranges))
    return;
  ranges.push_back(CreateCoreFileMemoryRange(region));
}
```
Above is the current version of this function, we just need to fix the first if statement to not just check if there are any permissions, but to check if the section is readable:
```
if (region.GetRange().GetByteSize() == 0 || ((region.GetLLDBPermissions() & lldb::ePermissionsReadable) == 0))
```

Reply:
This function should report any errors that are found when reading memory, but it should also save _any_ bytes that are read if `bytes_read` is not zero. So we might ask to read an entire region, but only get part of it for some reason, in that case we might get an error saying "error: not all bytes read from section", but if `bytes_read` is greater than zero, we need to save those bytes and not skip them.